src/actions/auth.js

```diff
@@ -0,0 +1,246 @@
+
+    return (dispatch) => {
+
+        const strangeLogin = internals.strangeActions.login({ email, password, token });
+
+        return dispatch(strangeLogin)
+
+        .then(() => {
+
+            return History.push('/dashboard');
+
+        })
+        .catch(::console.warn);
+    };
+};
+
+exports.logout = () => {
+
+    return (dispatch) => {
+
+        dispatch(internals.strangeActions.logout());
+        return History.push('/');
+    };
+};
+
+// Request Reset & Reset Password
+
+exports.requestResetRequest = () => {
+
+    return {
+        type: AuthAct.REQUEST_PASSWORD_RESET_BEGIN
+    };
+};
+
+exports.requestResetSuccess = () => {
+
+    return {
+        type: AuthAct.REQUEST_PASSWORD_RESET_SUCCESS
+    };
+};
+
+exports.requestResetFailure = (errMessage) => {
+
+    return {
+        type: AuthAct.REQUEST_PASSWORD_RESET_FAILURE,
+        payload: errMessage
+    };
+};
+
+exports.resetPasswordRequest = () => {
+
+    return {
+        type: AuthAct.RESET_PASSWORD_BEGIN
+    };
+};
+
+exports.resetPasswordSuccess = () => {
+
+    return {
+        type: AuthAct.RESET_PASSWORD_SUCCESS
+    };
+};
+
+exports.resetPasswordFailure = (errMessage) => {
+
+    return {
+        type: AuthAct.RESET_PASSWORD_FAILURE,
+        payload: errMessage
+    };
+};
+
+exports.requestPasswordReset = ({ email }) => {
+
+    return (dispatch) => {
+
+        dispatch(actions.requestResetRequest());
+
+        return WebClient.post('/users/request-reset', { email }, { responseType: 'text' })
+
+        .then(({ data, status }) => {
+
+            dispatch(actions.requestResetSuccess());
+            return History.push('/login');
+
+        })
+        .catch((err) => {
+
+            let errMessage = 'Unable to reset password. Please try again';
+
+            if (typeof err.response !== 'undefined') {
+                errMessage = err.response.data.message;
+            }
+
+            dispatch(actions.requestResetFailure(errMessage));
+        });
+    };
+};
+
+exports.resetPassword = (email, newPassword, resetToken) => {
+
+    return (dispatch) => {
+
+        dispatch(actions.resetPasswordRequest());
+
+        return WebClient.post('/users/reset-password', { email, newPassword, resetToken }, { responseType: 'text' })
+
+        .then(({ data, status }) => {
+
+            dispatch(actions.resetPasswordSuccess());
+            dispatch(actions.login({ email, password: newPassword }));
+
+        })
+        .catch((err) => {
+
+            let errMessage = 'Unable to reset password. Please try again.';
+
+            if (typeof err.response !== 'undefined') {
+                errMessage = err.response.data.message;
+            }
+
+            dispatch(actions.resetPasswordFailure(errMessage));
+        });
+    };
+};
+
+// StrangeAuth
+
+internals.strangeActions = StrangeAuth.makeActions({
+    login: ({ email, password, token }) => {
+
+        let authPromise;
+        let finalToken;
+
+        if (!!token) {
```

Comment:
`if (token)` would be OK too!

Reply:
Lmao that was my fault -- I was on a `!!` kick for a lil bit and we wrote this part together

Reply:
I remember that kick!  Casting to boolean is useful esp. when you want to return a boolean from a function for the consumer's benefit, or use a function that expects a boolean value.  We also use it inside react `render()` funcs because some falsy values actually could render!  For example, `0` is falsey but renders "0" into the dom, which could break something like `{userCount && <span>{userCount} users</span>}.

Reply:
👌 Noice.

Using booleans in conditionals is veeeery delicate at times